cookbook/form/data_transformers.rst

```diff
@@ -111,29 +111,100 @@ for converting to and from the issue number and the ``Issue`` object::
 Using the Transformer
 ---------------------
 
-Now that you have the transformer built, you just need to add it to your
-issue field in some form.
+As seen above our transformer requires an instance of an object manager. While for most
+use-cases using the default manager is fine we will let you pick the manager by it's name.
+In order to achieve this we will add a factory::
+
+    // src/Acme/TaskBundle/Form/DataTransformer/IssueToNumberTransformerFactory.php
+    namespace Acme\TaskBundle\Form\DataTransformer;
+
+    use Symfony\Bridge\Doctrine\ManagerRegistry;
+
+    class IssueToNumberTransformerFactory
+    {
+        /** 
+         * @var ManagerRegistry 
+         */
+        private $registry;
+
+        public function __construct(ManagerRegistry $registry)
+        {
+            $this->registry = $registry;
+        }
+
+        public function create($om)
+        {
+            return new IssueToNumberTransformer($this->registry->getManager($om));
+        }
+    }
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
+            acme_demo.factory.issue_transformer:
+                class: Acme\TaskBundle\Form\DataTransformer\IssueToNumberTransformerFactory
+                arguments: ["@doctrine"]
+                
+            acme_demo.type.task:
+                class: Acme\TaskBundle\Form\TaskType
+                arguments: ["@acme_demo.factory.issue_transformer"]
+
+    .. code-block:: xml
+
+        <service id="acme_demo.factory.issue_transformer" class="Acme\TaskBundle\Form\DataTransformer\IssueToNumberTransformerFactory">
```

Comment:
you should add the class attribute on a new line, to prevent horizontal scrolling (same below):

```xml
<service id="app.factory.issue_transformer"
    class="AppBundle\Form\DataTransformer\IssueToNumberTransformerFactory">
```